src/lib/device-types/storage/gh.ts

```diff
@@ -0,0 +1,42 @@
+import * as Octokit from '@octokit/rest';
+import * as _ from 'lodash';
+import * as Promise from 'bluebird';
+import { captureException } from '../../../platform/errors';
+
+import {
+	DEVICE_TYPE_REPO_NAME,
+	DEVICE_TYPE_REPO_OWNER,
+	DEVICE_TYPE_REPO_PATH,
+	DEVICE_TYPE_REPO_REF,
+} from '../../config';
+
+interface FetchDeviceTypeOptions {
+	owner: string;
+	repo: string;
+	path: string;
+	ref?: string;
+}
+
+export const fetchExternalDeviceTypes = (
+	options?: FetchDeviceTypeOptions,
+): Promise<string[]> => {
+	const realOptions = options || {
+		owner: DEVICE_TYPE_REPO_OWNER,
+		repo: DEVICE_TYPE_REPO_NAME,
+		path: DEVICE_TYPE_REPO_PATH,
+		ref: DEVICE_TYPE_REPO_REF,
+	};
+	const octokit = new Octokit();
+	return Promise.try(() => {
+		return octokit.repos.getContents(realOptions).then(dtSlugs => {
+			if (_.isArray(dtSlugs.data)) {
+				return dtSlugs.data.filter(e => e.type == 'dir').map(e => e.name);
+			}
+			const err = new Error(
+				`Unexpected response from GH: ${realOptions.owner}/${realOptions.repo}/${realOptions.path} should be a directory!`,
+			);
+			captureException(err, err.message);
+			return [];
```

Comment:
Is it fine that we don't throw the error here, and we resolve the promise instead?

Reply:
Yes, it will catch up later, but we might hit GH rate limits or similar.